python/cuspatial/cuspatial/core/interpolate.py

```diff
@@ -0,0 +1,195 @@
+        calls C++/Thrust code to compute cubic splines for each set of input
+        coordinates in parallel.
+        """
+        # error protections:
+        if len(t) < 5:
+            raise ValueError(
+                "Use of GPU cubic spline requires splines of length > 4"
+            )
+        if not isinstance(t, Series):
+            raise TypeError(
+                "Error: input independent vars must be cudf Series"
+            )
+        if not isinstance(y, (Series, DataFrame)):
+            raise TypeError(
+                "Error: input dependent vars must be cudf Series or DataFrame"
+            )
+        if not len(t) == len(y):
+            raise TypeError(
+                "Error: dependent and independent vars have different length"
+            )
+        if ids is None:
+            self.ids = Series([0, 0]).astype("int32")
+        else:
+            if not isinstance(ids, Series):
+                raise TypeError(
+                    "cuspatial.CubicSpline requires a cudf.Series"
+                )
+            if not ids.dtype == np.int32:
+                raise TypeError(
+                    "Error: int32 only supported at this time."
+                )
+            self.ids = ids
+        self.size = size if size is not None else len(t)
+        if not isinstance(self.size, int):
+            raise TypeError("Error: size must be an integer")
+        if not ((len(t) % self.size) == 0):
+            raise ValueError(
+                "Error: length of input is not a multiple of size"
+            )
+        if not isinstance(t, Series):
+            raise TypeError(
+                "cuspatial.CubicSpline requires a cudf.Series"
+            )
+        if not t.dtype == np.float32:
+            raise TypeError(
+                "Error: float32 only supported at this time."
+            )
+        if not isinstance(y, Series):
+            raise TypeError(
+                "cuspatial.CubicSpline requires a cudf.Series"
+            )
+        if not y.dtype == np.float32:
+            raise TypeError(
+                "Error: float32 only supported at this time."
+            )
+        self.t = t
+        self.y = y
+        if prefixes is None:
+            self.prefix = Series(
+                np.arange((len(t) / self.size) + 1) * self.size
+            ).astype("int32")
+        else:
+            if not isinstance(prefixes, Series):
+                raise TypeError(
+                    "cuspatial.CubicSpline requires a cudf.Series"
+                )
+            if not prefixes.dtype == np.int32:
+                raise TypeError(
+                    "Error: int32 only supported at this time."
+                )
+            self.prefix = prefixes
+
+        self.c = self._compute_coefficients()
+
+    def _compute_coefficients(self):
+        """
+        Utility method used by __init__ once members have been initialized.
+        """
+        if isinstance(self.y, Series):
```

Comment:
This and future patterns would be much cleaner in using the new `Frame` class which allows iterating over columns in general.